src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -92,6 +95,61 @@ protected NUnitTestAdapter()
 
         #region Protected Helper Methods
 
+        // The Adapter is constructed using the default constructor.
+        // We don't have any info to initialize it until one of the
+        // ITestDiscovery or ITestExecutor methods is called. Each
+        // Discover or Execute method must call this method.
+        protected void Initialize(IDiscoveryContext context)
+        {
+            var settingsXml = context?.RunSettings?.SettingsXml;
+            if (string.IsNullOrEmpty(settingsXml))
+                settingsXml = "<RunSettings />";
```

Comment:
I'd prefer a short-circuit here rather than parsing a dummy XML document.

Reply:
I just copied nunit3 adapter code.

Reply:
Ah, I see: https://github.com/nunit/nunit3-vs-adapter/blob/master/src/NUnitTestAdapter/AdapterSettings.cs#L177

That does make sense; you might want to apply a series of default values without duplicating code. :+1: